src/NUnitFramework/framework/Attributes/DatabaseDataSourceAttribute.cs

```diff
@@ -0,0 +1,161 @@
+﻿
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal.Builders;
+using System.Collections;
+using NUnit.Framework.Internal;
+
+namespace NUnit.Framework.Attributes
+{
+    /// <summary>
+    /// Specifies an external data source for test methods. This is evaulated at compile time.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = true, Inherited = false)]
+    class DbDataAttribute : NUnitAttribute, ITestBuilder, IImplyFixture
```

Comment:
Class name should match the filename and I prefer the full `DatabaseDataSourceAttribute`.